athena-scripts/raw_stage_optimisation_solution/scripts/core_preprocessing_functions.py

```diff
@@ -166,14 +167,34 @@ def generate_raw_select_filter(json_data, database, table, filter_processed_dt):
         if event_processing_selection_criteria_limit is None:
             raise ValueError("filter value for event_processing_testing_criteria is not found within config rules")
         print(f'config rule: event_processing_testing_criteria | filter: {event_processing_testing_criteria_filter}')
-
-        sql=f'''select * from \"{database}\".\"{table}\"'''
-
+        
+        deduplicate_subquery = f'''select *,
+			            row_number() over (
+				            partition by event_id
+				            order by cast(
+                                        concat(
+                                            cast(year as varchar),
+                                            cast(lpad(cast(month as varchar), 2, '0') as varchar), 
+                                            cast(lpad(cast(day as varchar), 2, '0') as varchar)
+                                        ) as int
+                                    ) desc
+			                ) as row_num
+               		    from \"{database}\".\"{table}\" as t '''
+                     
+        sql = 'select * from \"{database}\".\"{table}\"'
+        
         if event_processing_testing_criteria_enabled and event_processing_testing_criteria_filter is not None:
-            sql = sql + f' where {event_processing_testing_criteria_filter}'
+            deduplicate_subquery = deduplicate_subquery + f'where {event_processing_testing_criteria_filter}'
```

Comment:
```suggestion
            deduplicate_subquery = deduplicate_subquery + f' where {event_processing_testing_criteria_filter}'
```
do you need a space here?